### PR TITLE
Add feature store lock/unlock implementation

### PR DESCRIPTION
**Pull request by ddl-joyce-zhao**
## Description

added feature store lock/unlock implementation.

## Related Issue

https://dominodatalab.atlassian.net/browse/DOM-42243

## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [x] 🥂 Improvement (non-breaking change which improves an existing feature)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

## Checklist

- [x] I've read the [`CONTRIBUTING.md`](https://github.com/dominodatalab/domino-data/blob/main/CONTRIBUTING.md) guide.
- [x] I've updated the code style using `make codestyle`.
- [x] I've written tests for all new methods and classes that I created.
- [x] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

File: domino_data/_feature_store/run.py [hunk 1]
```diff
@@ -31,10 +31,19 @@
...
     )
+
+    parser.add_argument(
+        "-r",
+        "--retry",
+        metavar="MAX_LOCK_RETRY_COUNT",
+        type=int,
+        required=False,
+        help="the maximum feature store lock retry count",
+    )
```
Comment (ddl-catherinetu) [inline domino_data/_feature_store/run.py:42]: so this argument would be used by the job script right? does this mean the user can somehow set how many retries they'd want for feature store syncs? where would they set that?

File: domino_data/_feature_store/sync.py [hunk 2]
```diff
@@ -168,17 +193,22 @@ def feature_store_sync(
     if not branch_name:
         branch_name = repo.active_branch.name
 
+    if not max_retries:
+        max_retries = 60
```
Comment (ddl-catherinetu) [inline domino_data/_feature_store/sync.py:197]: wow that's a big number :D

File: domino_data/_feature_store/run.py (hunk 1 above)
Comment (ddl-joyce-zhao) [reply domino_data/_feature_store/run.py:42]: This script is not supposed to be called by the users but by our front end. That's why we make the package name with the starting "_" to indicate this is something private. 

When the front end starts a job to do the synchronization, it's supposing to call the script 
"python3 -m domino_data.feature_store.run_featurestore_sync -r <max-retries>"

This is an optional argument, if not provided, it falls to the default 60. Basically retrying for 1 minute.

File: domino_data/_feature_store/sync.py (hunk 2 above)
Comment (ddl-joyce-zhao) [reply domino_data/_feature_store/sync.py:197]: I simply give one minute for the users to try the locking. Do you have a good sense of how long waiting time is acceptable?

Comment (ddl-catherinetu) [reply domino_data/_feature_store/sync.py:200]: oh is this 60 number the number of seconds? I thought it was number of retries

File: domino_data/_feature_store/run.py (hunk 1 above)
Comment (ddl-catherinetu) [reply domino_data/_feature_store/run.py:42]: got it!

**Review by ddl-catherinetu — APPROVED** [review]

File: domino_data/_feature_store/sync.py (hunk 2 above)
Comment (ddl-joyce-zhao) [reply domino_data/_feature_store/sync.py:200]: It's the number of retries. After each retry, it sleeps for 1 second. So basically retry for 1 minute.

**Pull request merged**